deploy/basic/terraform/dns.tf

```diff
@@ -0,0 +1,51 @@
+resource "oci_dns_zone" "mushop_dns_zone" {
+  # Conditional
+  count = var.enable_dns_zone ? 1 : 0
+
+  #Required
+  compartment_id = var.lb_compartment_ocid
+  name = var.dns_zone_name
+  zone_type = "PRIMARY"
+
+}
+
+resource "oci_dns_rrset" "mushop-internal" {
+  # Conditional
+  count = var.enable_dns_zone ? 1 : 0
```

Comment:
I recommend that you resolve this in to a local, based on the `var.enable_dns_zone` and `var.use_only_always_free_eligible_resources`. The stack has a feature to let people use only always free resources when deploying, which this might break.

Reply:
The complete MuShop is already being updated to include some or all of these features, via a different mechanism, and the basic one that is not Kubernetes might be easier to digest.  This is essentially adding the functionality that @junior added to the diagram within the MSZ readme.  

Regarding ORM, I will re-test, and might have to do something differently if ACME isn't going to work.  A shame, because it does go nicely in with OCI DNS and validation of domain ownership.  If the cert cannot go in, then the listener and WAF are harder to use.

Agreed on the offline activity - I will work on the documentation to show how this is required and what to do.

> This is an excellent set of feature additions. Couple of questions to consider.
> 
> * Why make these changes on the Basic version ? The basic version needs to support the Always Free tier and all features that are not in the Always Free needs to have toggles triggered by the high level option to simply `Use Always Free Only`.
> * The primary and recommended way for customers to deploy this is through ORM, so the providers we use needs to be supported ones.
> * There are several "Offline" activities (DNS setup etc.) that don't fit well within the ORM stack deployment model

Reply:
You can request new providers on the ORM, that need to be validated and approved by ORM team and Corp Arch. You cannot have an unsupported provider on the basic stack or will break everything else, even if was supposed to only be used from Terraform and not ORM. Currently Terraform does not support conditional provider.

You have an option to create a fork with that solution, or include the optional instructions on the wiki.